iModelCore/GeomLibs/geom/test/SerializationTest/t_imodeljson.cpp

```diff
@@ -121,44 +111,62 @@ TEST(IModelJson,BytesToXXX)
 +---------------+---------------+---------------+---------------+---------------+------*/
 TEST(IModelJson,ReadFiles)
     {
-    for (auto & name :
-        bvector<CharCP> {
-            "arc1",
-            "lineSegment5",
-            "arc",
-            "lineSegment",
-            "lineString",
-            "pointString",
-            "path",
-            "loop",
-            "parityRegion",
-            "bcurve",
-            "cylinder",
-            "cone",
-            "box",
-            "sphere",
-            "torusPipe",
-            "linearSweep",
-            "rotationalSweep",
-            "ruledSweep",
-            "bsurf",
-            "bsurfWithBoundary",
-            "indexedMesh",
-            })
+    BeFileName directoryPath;
+    BeTest::GetHost().GetDocumentsRoot(directoryPath);
+    directoryPath.AppendToPath(L"GeomLibsTestData");
+    directoryPath.AppendToPath(L"IModelJson");
+    WCharCP dirPath = directoryPath.GetName();
+    for (const auto &entry : std::filesystem::directory_iterator(dirPath))
         {
+        BeFileName filePath = BeFileName(entry.path().c_str());
+        WString fileNameStr;
+        filePath.ParseName(NULL, NULL, &fileNameStr, NULL);
+        WCharCP fileName = fileNameStr.c_str();
         bvector<IGeometryPtr> geometry;
-        WString wName;
-        wName.AppendA(name);
-        if (Check::True(ReadIModelJson (geometry, L"IModelJson", L"IModelJsonFromNative", wName.c_str(), nullptr, L"imjs", nullptr, L"fbjs"), "read imjs file"))
+        if (Check::True(ReadIModelJson (
+            geometry, L"IModelJson", L"IModelJsonFromNative", fileName, nullptr, L"imjs", nullptr, L"fbjs"),
+            "read imjs file"))
             {
-            for (auto const& g : geometry)
+            for (IGeometryPtr const& g : geometry)
                 {
                 if (Check::True(g.IsValid(), "Imported geometry is valid"))
-                    Check::NearRoundTrip(*g, 0.0, name);
+                    Check::NearRoundTrip(*g, 0.0, (char*) fileName);

```

Comment:
This is not good practice, even if it works in this case. We are stuck with a plethora of char* types in the geomlibs test API, but this type should be avoided in general, in favor of WCharCP (defined to be wchar_t const*) for representing strings. A better solution would be to create an overload of `NearRoundTrip` that takes WCharCP; then the existing method can call the new overload passing in `WString(pString, BentleyCharEncoding::Locale).c_str()`. Otherwise, if you keep the char* cast, you should at least warn about the bad practice with a comment.

Reply:
As discussed will address this in the next PR.

Reply:
Due to too many merge conflicts, I decided to delay the whiteSpace PR. Therefore, I want to address this comment in this PR.

Because we are allowed to change imdoel-native and imodel02 APIs, I will update the existing API to take `WCharCP` instead of `char const*` in these 2 repos. For PPBase I will deprecate the old API and add a new overload. Note that there is only 1 other call to `Check::NearRoundTrip` in imodel-native geomlibs.

Please let me know if this is a good plan.

Reply:
Sounds good.  Actually, I'd just change the arg to WCharCP in _both_ repos, since I don't think the Check class is public API (being in test directory).

Reply:
I need to discuss this with you. If we change API in imodel-native and imodel02 but deprecate in PPBase we end up having different names in different repos for API with the same behaviour. So I was thinking even if we dont deprecate in imodel-native and imodel02, we should use the same API name as PPBase.

Reply:
Yes, same names for `Check::NearRoundTrip` in all 3 native repos. No deprecation needed for changes to `Check` since it isn't public API.

Reply:
I dont meant just `Check::NearRoundTrip`
I meant all other functions that we changes line `BytesToGeometry` or `BytesToCurvePrimitive`
I dont think it makes sense to have `BytesToGeometry` in 2 repos and then `BytesToGeometrySafe` in another repo when both do the same exact thing.

Reply:
Those methods are in the PPBase public API, unfortunately.

Reply:
Yes, I know. I mean in imodel02 and imodel-native, rather than changing the body and keeping the API name, we also change the API name to match with PPBase. For example, in PPBase the new method would be called `BytesToGeometrySafe` so in imodel02 and imodel-native we also should call the safe method `BytesToGeometrySafe` and not `BytesToGeometry` to be consistent with PPBase.

Reply:
Ah, I see. I guess since all three native repos are going forward for the time being, it would be nice for their APIs to be the same.  How about this: add simple "Safe"-suffixed wrapper methods to imodel02/imodel-native that just call the updated `BytesToGeometry` methods. This way there is still a 1-1 correspondence of methods in the FB APIs of the three repos.  Callers in imodel02/imodel-native can continue to use the newly updated BytesToGeometry methods; no imodel02/imodel-native methods will actually call the new "Safe" wrappers, but this will make it apparent to which imodel02/imodel-native methods the PPBase "Safe" methods map. You don't even have to put the new "Safe" wrappers in the imodel02/imodel-native API .h file. You might want to write a test that covers them, and verifies that they produce the same output as the methods they wrap.

Reply:
yes, I like this idea. I will add the safe wrappers. 

FYI, I replaced `char const*` with `WCharCP` in `Check::NearRoundTrip` args so I can resolve this conversation.